netbox/utilities/forms/fields/dynamic.py

```diff
@@ -43,7 +43,7 @@ def get_bound_field(self, form, field_name):
 
         if data is not None:
             self.choices = [
-                choice for choice in self.choices if choice[0] in data
+                choice for choice in self.choices if isinstance(self.choices, str) if choice[0] in data
```

Comment:
```suggestion
                choice for choice in self.choices if isinstance(self.choices, str) and choice[0] in data
```

I think `and` should suffice here